### PR TITLE
Validation alerts

### PR DESCRIPTION
**Pull request by cfarm**
Alert message styles for field-level errors, form-level errors, and form-level success for validation forms.

Alerts will be refined in future PR - this is just implementing the basic styling from cfgov-refresh.
## Additions
- background color variables for form alerts
- new `.cf-notification` element for form-level messaging - this will be expanded in future pr as needed to follow the new standards.
- `.cf-notification` source: https://github.com/cfpb/cfgov-refresh/blob/5a1c9afdec01e3bec31fc55935c1103deae1a822/cfgov/v1/preprocessed/css/cf-notifications.less
## Changes
- icon styles and documentation
## Review
- @Scotchester 
## Screenshots

screen shot 2015-09-28 at 1 26 32 pm
## Todos
- @virginiacc will refine these styles for the new standards in a future PR
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [ ] Passes all existing automated tests
- [ ] New functions include new tests
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged
- [ ] Visually tested in supported browsers and devices 
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by Scotchester** [comment]
I think we want this to be its own component outside of cf-forms.

@jimmynotjim Can you comment on the status of [jimmynotjim/cf-notifications:initial-dev](https://github.com/jimmynotjim/cf-notifications/tree/initial-dev)?

**Comment by cfarm** [comment]
@Scotchester OK, if we end up needing to do that then @virginiacc will be taking that on.

**Comment by virginiacc** [comment]
@Scotchester @cfarm The main change in the new alerts standards seems to be the colors used, so it should be easy to make those updates either to the code here or in the new cf-notifications repo. That said, since we're looking at wrapping up the related design manual update in the next day or so, it seems like the simplest course might be to update the alerts here and transfer them over to the other repo when time allows. Unless the new repo is about ready to go?

**Comment by jimmynotjim** [comment]
@Scotchester it works, it's a clone of the v1 notifications, but I never got around to refactoring the JS, testing it, and getting it building in the docs.

As far as where it lives, I had originally made it separate because there are some notifications that aren't injected into the page, like when you filter posts http://localhost:8000/events/archive/?filter_range_date_gte=&filter_range_date_lte=&filter_tags=Regulations and I think we wanted to leave it open to using them in the beta banner.

**Comment by Scotchester** [comment]
Ok, I'm fine with moving forward with them in this repo for now, but I think we definitely want them to eventually live in their own repo, since they're general-purpose and not specific to forms.

All references to CFPB color variable should be converted to semantically-named color variables that can be overridden. You can probably easily copy from what Jimmy's done on his branch: https://github.com/jimmynotjim/cf-notifications/blob/initial-dev/src/less/cf-notifications.less

File: src/cf-forms.less [hunk 1]
```diff
@@ -448,6 +416,86 @@ textarea {
...
+        &__warning {
+            color: #ff931b;
+        }
+    }
+
+    &_text {
+        .h4();
+        margin: 0;
+
+        .short-desc {
```
Comment (jimmynotjim) [inline src/cf-forms.less:514]: I forget if this is even necessary. I made a quick look through v1and I can't find an instance of the notifications that have a `.short-desc` child-block inside the `_text` block. Could probably be removed for now and re-added if necessary later.

File: src/cf-forms.less [hunk 2]
```diff
@@ -448,6 +416,86 @@ textarea {
...
+    &_icon {
+        position: absolute;
+        top: @notification-padding__px + 2px;
+        left: @notification-padding__px + 4px;
+
+        color: @gray-20;
+        font-size: unit( 18px / @base-font-size-px, em );
+
+        &__error {
+            color: #d14327;
```
Comment (jimmynotjim) [inline src/cf-forms.less]: These absolute colors came from @duelj as slight tweaks to the input colors. Justin do you remember what tweaking we did? Could we automate it somehow using the input variables? Ex. `darken(@input-error, 15%)` or something similar using Less color functions.

File: src/cf-forms.less [hunk 3]
```diff
@@ -448,6 +416,86 @@ textarea {
...
+
+    background: @gray-20;
+    border: 1px solid @gray;
+
+    &__visible {
+      display: block;
+    }
+
+    &__error {
+        background: @redorange-20;
```
Comment (jimmynotjim) [inline src/cf-forms.less]: We should probably replace all of these background colors with variables that can then be over-written in the overrides file.

File: src/cf-forms.less (hunk 2 above)
Comment (Scotchester) [inline src/cf-forms.less]: Bump for @duelj input.

File: src/cf-forms.less (hunk 3 above)
Comment (Scotchester) [inline src/cf-forms.less]: Yep. Reposting my general comment from earlier:

All references to CFPB color variable should be converted to semantically-named color variables that can be overridden. You can probably easily copy from what Jimmy's done on his branch: https://github.com/jimmynotjim/cf-notifications/blob/initial-dev/src/less/cf-notifications.less

**Comment by cfarm** [comment]
I updated the color variables, here's a new screenshot (with warning alert message too)

screen shot 2015-10-13 at 1 26 08 pm

@jimmynotjim I'm including the `.short-desc` style since we are using it in our Design Manual examples for the list text (see http://cfarm.github.io/design-manual/ui-toolkit/form-fields.html for the elements in the wild).

**Comment by jimmynotjim** [comment]
Ah right, there are times it's more than just the message. I think this is good now :+1: cc @Scotchester

**Comment by Scotchester** [comment]
Obsolete.

**Pull request closed without merging**